queries/nix.scm

```diff
@@ -0,0 +1,312 @@
+;;
+;; Statements
+;;
+
+;; Any top-level expression that isn't a comment is a statement
+(
+    (source_code
+        (_) @statement
+    ) @_.iteration
+    (#not-type? @statement comment)
+)
+(source_code) @comment.iteration
+
+;; Any foo = <expression> anywhere is a statement
+[
+    (inherit)
+    (binding)
+] @statement
+(binding_set) @map.iteration @list.iteration @ifStatement.iteration
+
+;; This matches assert statements as stand-alone statements. This
+;; is because it's common to have a series of assert statements in a file, which
+;; would otherwise all match as a single statement. See the playground/nix/asserts.nix
+;; file for an example.
+(
+    (assert_expression
+        "assert" @statement.start
+        condition: (_)
+        ";" @statement.end
+        body: (_)
+    ) @_.domain
+    (#not-parent-type? @_.domain binding)
+)
+(assert_expression
+    "assert" @functionCallee @functionCall.start
+    condition: (_) @condition
+    body: (_) @branch @functionCall.end.startOf
+) @_.domain
+
+(
+    (with_expression) @statement
+    (#not-parent-type? @statement binding)
+)
+
+;;!! let
+;;!!   a = 1;
+;;!   <******
+;;!!   b = 2;
+;;!    *****>
+;;!! in a + b
+(let_expression
+    "let"
+    (_) @statement.iteration @value.iteration @name.iteration
+)
+
+;;
+;; Conditionals
+;;
+
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^^^^^^^^
+(if_expression) @ifStatement @branch.iteration @condition.iteration
+;;!! key = if a then b else c;
+;;!        ^^^^^^^^^^^
+;;!        xxxxxxxxxxxx
+(if_expression
+    "if" @branch.start
+    condition: (_)
+    "then"
+    consequence: (_) @branch.end
+)
+;;!! key = if a then b else c;
+;;!                    ^^^^^^
+;;!                   xxxxxxx
+(if_expression
+    "else" @branch.start
+    alternative: (_) @branch.end
+)
+
+;;!! key = if a > 10 then b else c;
+;;!           ^^^^^^
+(if_expression
+    condition: (_) @condition
+) @_.domain
+
+;;
+;; Lists and maps
+;;
+
+;;!! foo = [ a b c ];
+;;!          ^
+;;!          xx
+(list_expression
+    element: (_) @collectionItem
+)
+
+;;!! foo = [ a b c ];
+;;!        ^^^^^^^^^
+(list_expression
+    element: (_)+ @list.interior
```

Comment:
Hmm I get an error message

```
A capture with the same name may only appear once in a single pattern: list.interior 1:9-1:12,list.interior 1:13-1:16
```

If I say `"clear inside list"` with the following code:

```nix
{
    a = ["0" "1"];
}
```

Reply:
This is fixed. I had to switch to `interior.starting` and `interior.ending` to get rid of it, but didn't look into why.